src/InteractiveWindow/Editor/History.cs

```diff
@@ -159,11 +164,16 @@ private Entry Get(string pattern, int step)
 
         internal Entry GetNext(string pattern = null)
         {
+            //if you are at the last history entry or history navigation pointer is
+            //not initialized then return null as there is no next entry
+            if ((_current == Length - 1) || (_current == -1)) return null;
             return Get(pattern, step: +1);
         }
 
         internal Entry GetPrevious(string pattern = null)
```

Comment:
Not part of this change but the optional parameter value is not necessary in GetPrevious or GetNext.